docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,322 @@
+
+- Use xargs utility on Unix
+
+```bash
+oras push $REGISTRY/$REPO:$TAG hello.txt --format '{{.Ref}}' |\
+xargs -I _ oras attach --artifact-type sbom/example _ sbom.spdx --format '{{.Ref}}' |\
+xargs -I _ notation sign _ 
+```
+
+- Use environmental variables on Unix
+
+```bash
+REFERENCE_A=$(oras push $REGISTRY/$REPO:$TAG hello.txt --format '{{.Ref}}')
+REFERENCE_B=$(oras attach --artifact-type sbom/example $REFERENCE_A sbom.spdx --format '{{.Ref}}') 
+notation sign $REFERENCE_B
+```
+
+- Use [ConverFrom-Json](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json) on Windows PowerShell
+
+```powershell
+$REFERENCE_A=oras push $REGISTRY/$REPO:$TAG hello.txt --format json --no-tty | ConvertFrom-Json
+$REFERENCE_B=oras attach --artifact-type sbom/example $REFERENCE_A.Ref sbom.spdx --format json --no-tty | ConvertFrom-Json
+notation sign $REFERENCE_B.Ref
+```
+
+### CI/CD
+
+Bob is a DevOps engineer. He uses the ORAS GitHub Actions [Setup action](https://github.com/oras-project/setup-oras) to install ORAS in his CI/CD workflow. He wants to chain multiple ORAS commands in a Shell script to perform multiple operations.
+
+For example, pull multiple files (layers) from a repository and filter out the file path of its first layer in the standard output, then pass the pulled first layer to the second command `docker import` for further operation. 
+
+```yaml
+jobs:
+  example-job:
+    steps:
+      - uses: oras-project/setup-oras@v1
+      - run: |
+          PATH=`oras pull $REGISTRY/$REPO:$TAG --format '{{.first .Files.Path}}'`
+          docker import $PATH
+```
+
+## Proposal and desired user experience
+
+Enable users to use the `--format <type>` flag to format output into structured data (e.g. JSON) and use the `--format` with [Go template](https://pkg.go.dev/text/template) to manipulate the output fields. Users can still use `--format '{{toRawJson .}}'` to get raw JSON output.
+
+### oras pull 
+
+Pull a repository and display its metadata as formatted JSON in standard output. The following fields should be formatted in a JSON output:
+
+- `Ref`: full artifact reference by digest, e.g, `$REGISTRY/$REPO@$DIGEST`
+-  `Files`: a list of downloaded files
+    -  `Path`: the absolute file path of the pulled file (layer)
+    -  `Ref`: full reference by digest of the pulled file (layer)
+    -  `MediaType`: media type of the pulled file (layer) 
+    -  `Digest`: digest of the pulled file (layer) 
+    -  `Size`: file size in bytes
+
+Pull a repository that contains multiple layers (files) and show their descriptor metadata as pretty JSON in standard output.
```

Comment:
What signifies the output to be "pretty" in the following command?

Reply:
If user specifies the flag `--format json`, the output will be "pretty" in a structured JSON format

Reply:
I updated this section to avoid ambiguous description.